app/upgrades/v14/upgrades.go

```diff
@@ -252,6 +285,39 @@ func MigrateAccount1(ctx sdk.Context, evk evmosvestingkeeper.Keeper, sk stakingk
 	return nil
 }
 
+// Migrate the stakeibc params, specifically:
+//   - Remove SafetyNumValidators
+//   - Remove SafetyMaxSlashPercentage
+//   - Add ValidatorSlashQueryThreshold
+//
+// NOTE: If a parameter is added, the old params cannot be unmarshalled
+// to the new schema. To get around this, we have to set each parameter explicitly
+// Considering all mainnet stakeibc params are set to the default, we can just use that
+func MigrateStakeibcParams(ctx sdk.Context, k stakeibckeeper.Keeper) {
+	params := stakeibctypes.DefaultParams()
+	k.SetParams(ctx, params)
+}
+
+// Since the Query struct was updated, it's easier to just clear out any pending
+// queries rather than attempt to migrate them
+func ClearPendingQueries(ctx sdk.Context, k icqkeeper.Keeper) {
+	for _, query := range k.AllQueries(ctx) {
+		k.DeleteQuery(ctx, query.Id)
+	}
+}
+
+// Enable LSM liquid stakes for Gaia
+func EnableLSMForGaia(ctx sdk.Context, k stakeibckeeper.Keeper) error {
+	hostZone, found := k.GetHostZone(ctx, GaiaChainId)
+	if !found {
+		return stakeibctypes.ErrHostZoneNotFound.Wrapf(GaiaChainId)
+	}
+
+	hostZone.LsmLiquidStakeEnabled = true
```

Comment:
LSM will be turned on on Stride before it's live on the Hub. Thinking this through: in theory Stride will be willing to accept LSM IBC vouchers for LSMLiquidStake as soon as this upgrade goes live. But no such vouchers will exist, so LSMLiquidStakes will all fail until Hub has LSM enabled. Is that right?

Reply:
exactly!